docs/src/main/asciidoc/hibernate-orm-panache.adoc

```diff
@@ -1,326 +1,752 @@
+----
+** Repository pattern:
++
+[source,java]
+----
+List<Person> livingPersons = personRepository.list("status", Status.Alive);
+----
 
-// finding all living persons
-List<Person> livingPersons = Person.list("status", Status.Alive);
+* Count all persons:
+** Active record pattern:
++
+[source,java]
+----
+long countAll = person.count();
+----
+** Repository pattern:
++
+[source,java]
+----
+long countAll = personRepository.count();
+----
 
-// counting all persons
-long countAll = Person.count();
+* Count all living persons:
+** Active record pattern:
++
+[source,java]
+----
+long countAlive = person.count("status", Status.Alive);
+----
+** Repository pattern:
++
+[source,java]
+----
+long countAlive = personRepository.count("status", Status.Alive);
+----
 
-// counting all living persons
-long countAlive = Person.count("status", Status.Alive);
+* Delete all living persons:
+** Active record pattern:
++
+[source,java]
+----
+person.delete("status", Status.Alive);
+----
+** Repository pattern:
++
+[source,java]
+----
+personRepository.delete("status", Status.Alive);
+----
 
-// delete all living persons
-Person.delete("status", Status.Alive);
 
-// delete all persons
-Person.deleteAll();
+* Delete all persons:
+** Active record pattern:
++
+[source,java]
+----
+person.deleteAll();
+----
+** Repository pattern:
++
+[source,java]
+----
+personRepository.deleteAll();
+----
 
-// update all living persons
-Person.update("name = 'Moral' where status = ?1", Status.Alive);
+* Delete by ID:
+** Active record pattern:
++
+[source,java]
+----
+boolean deleted = person.deleteById(personId);
+----
+** repository pattern:
++
+[source,java]
+----
+boolean deleted = personRepository.deleteById(personId);
+----
 
+* Set the name of all living persons to Mortal:
+** Active record pattern:
++
+[source,java]
+----
+person.update("name = 'Mortal' where status = ?1", Status.Alive);
+----
+** Repository pattern:
++
+[source,java]
+----
+personRepository.update("name = 'Mortal' where status = ?1", Status.Alive);
 ----
 
-All `list` methods have equivalent `stream` versions.
+All `List` methods have equivalent `stream` methods, which requires a transaction to work as shown in the following example:
 
 [source,java]
 ----
-Stream<Person> persons = Person.streamAll();
+Stream<Person> persons = personRepository.streamAll();
 List<String> namesButEmmanuels = persons
     .map(p -> p.name.toLowerCase() )
     .filter( n -> ! "emmanuel".equals(n) )
     .collect(Collectors.toList());
 ----
 
-NOTE: The `stream` methods require a transaction to work.
+== Hibernate ORM with Panache queries
+
+Hibernate Query Language (HQL) is a query language similar to Structured Query Language (SQL). However, HQL works with persistent objects and properties instead of tables and columns. For more information about HQL, see the "HQL and JPQL" section of the xref:https://docs.jboss.org/hibernate/orm/5.4/userguide/html_single/Hibernate_User_Guide.html[_Hibernate ORM User Guide_].
```

Comment:
I feel like this belongs closer to where we define "Simplified Hibernate ORM with Panache queries" below, no? This seems randomly placed.

Reply:
This is an introduction section for the whole query subject. For the order, I followed the order in the source. So now I've moved 'Simplified' to below the intro.

Reply:
OK fine, good move, thanks.